zha/application/platforms/sensor/__init__.py

```diff
@@ -202,14 +202,18 @@ def _init_from_quirks_metadata(self, entity_metadata: ZCLSensorMetadata) -> None
             self._divisor = entity_metadata.divisor
         if entity_metadata.multiplier is not None:
             self._multiplier = entity_metadata.multiplier
+        if entity_metadata.translation_key is not None:
```

Comment:
I think there is a restriction on translation key too if device class is set. I'll peek at core when I have a chance to validate this.

Reply:
I don't think it's a restriction in HA. For some device classes (like battery), there's just a standard name, so we don't need to set the name (or translation key) again if we want to use that default.

Reply:
However, the translation key stuff should likely be changed here, not in this sensor class:
https://github.com/zigpy/zha/blob/e1ae71cc31cd53700038881de1f4ee51586caca9/zha/application/platforms/__init__.py#L347-L349
I think we can just remove the device class translation key check. There are cases where we want to use a custom name (overriding the built-in/default name for the device class). (We already do for some ZHA sensors)